ephios/event_management/views.py

```diff
@@ -134,6 +141,82 @@ def get_queryset(self):
+    model = Event
+    template_name = "event_management/event_copy.html"
+    form_class = EventDuplicationForm
+
+    def setup(self, request, *args, **kwargs):
+        super().setup(request, *args, **kwargs)
+        self.object = self.get_object()
+
+    def form_valid(self, form):
+        occurences = form.cleaned_data["recurrence"].between(
+            datetime.now() - timedelta(days=1),
+            datetime.now() + timedelta(days=365),
+            inc=True,
+            dtstart=datetime.combine(
+                DateField().to_python(self.request.POST["start_date"]), datetime.min.time()
+            ),
+        )
+        for date in occurences:
+            event = self.get_object()
+            start_date = event.get_start_time().date()
+            shifts = event.shifts.all()
+            event.pk = None
+            event.save()
+            assign_perm(
+                "view_event", get_groups_with_perms(self.get_object(), ["view_event"]), event
+            )
+            assign_perm(
+                "change_event", get_groups_with_perms(self.get_object(), ["change_event"]), event
+            )
+
+            for shift in shifts:
+                shift.pk = None
+                # shifts on following days should have the same offset from the new date
+                offset = shift.start_time.date() - start_date
+                # shifts ending on the next day should end on the next day to the new date
+                end_offset = shift.end_time.date() - shift.start_time.date()
+                shift.end_time = datetime.combine(
+                    date.date() + offset + end_offset, shift.end_time.time()
+                )
+                shift.meeting_time = datetime.combine(
+                    date.date() + offset, shift.meeting_time.time()
+                )
+                shift.start_time = datetime.combine(date.date() + offset, shift.start_time.time())
+                shift.event = event
+                shift.save()
+                event.shifts.add(shift)
+
+        messages.success(self.request, _("Event succesfully copied."))
+        return redirect(reverse("event_management:event_list"))
+
+
+class RRuleOccurenceView(CustomPermissionRequiredMixin, View):
+    permission_required = "event_management.add_event"
+
+    def post(self, *args, **kwargs):
+        try:
+            recurrence = RecurrenceField().clean(self.request.POST["recurrence_string"])
+            return HttpResponse(
+                json.dumps(
+                    recurrence.between(
+                        datetime.now() - timedelta(days=1),
+                        datetime.now() + timedelta(days=365),
+                        inc=True,
+                        dtstart=datetime.combine(
+                            DateField().to_python(self.request.POST["dtstart"]), datetime.min.time()
+                        ),
+                    ),
+                    cls=CustomJSONEncoder,
+                )
```

Comment:
You should format the dates at some point, either here or in JS:

![image](https://user-images.githubusercontent.com/5970076/97103670-3852c080-16ae-11eb-83a4-037b403e011d.png)